### ISSUE TITLE
Re-add Missing Modules

### ISSUE THREAD

**Issue by voetberg** [deepskies/DeepAstroUDA#8]
There are some missing modules that would make it impossible to run this code. 

Ones documented are: 

- LinearAverage in `models`
- `mydataset` and  `ImageFolder`
- `UnalignedDataLoader`
- `Inference` (in general)

**Comment by AleksCipri** [comment]
I found a missing loss function. So that is added. I'll try to figure out the rest...but I doubt that I'll be able to resolve all of this without Ashia :(

**Comment by AleksCipri** [comment]
Also added models/LineareAverage.py from Wu et al 2018.
https://github.com/zhirongw/lemniscate.pytorch/blob/master/lib/LinearAverage.py

**Comment by AleksCipri** [comment]
ImageFolder and UnalignedDataLoader I found here:
https://github.com/zhengqili/unsupervised-learning-intrinsic-images

I hope nothing is modified compared to the original code, but I feel that at least ImageFolder is likely modified in some way...
